src/Core/Utilities/AccountManagement/AccountsManager.cs

```diff
@@ -100,8 +102,15 @@ await _vaultTimeoutService.ShouldLockAsync())
                 {
                     _accountsManagerHost.Navigate(NavigationTarget.AddEditCipher);
                 }
-                else if (Options.FromFido2Framework)
+                else if (Options.FromFido2Framework || fido2MakeCredentialConfirmationUserInterface.IsConfirmingNewCredential)
                 {
+                    if (fido2MakeCredentialConfirmationUserInterface.IsConfirmingNewCredential)
+                    {
+                        // If we are already confirming a credential we don't need to navigate again.
+                        // This could happen when switching accounts for example.
+                        return;
+                    }
+
```

Comment:
🎨 To simplify this you can write it as:
```suggestion
                else if (fido2MakeCredentialConfirmationUserInterface.IsConfirmingNewCredential)
                {
                    // If we are already confirming a credential we don't need to navigate again.
                    // This could happen when switching accounts for example.
                    return;
                }
                else if (Options.FromFido2Framework)
                {
                    
```